client/go/internal/vespa/deploy.go

```diff
@@ -119,6 +119,126 @@ func ZoneFromString(s string) (ZoneID, error) {
 	return ZoneID{Environment: parts[0], Region: parts[1]}, nil
 }
 
+func Fetch(deployment DeploymentOptions, path string) (string, error) {
+	if util.IsDirectory(path) {
+		path = filepath.Join(path, "application.zip")
+	}
+	if util.PathExists(path) {
+		return "", fmt.Errorf("%s already exists", path)
+	}
+	if deployment.Target.IsCloud() {
+		return path, fetchFromController(deployment, path)
+	}
+	return path, fetchFromConfigServer(deployment, path)
+}
+
+func deployServiceGet(url string, deployment DeploymentOptions, w io.Writer) error {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	response, err := deployServiceDo(req, 0, deployment)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+	_, err = io.Copy(w, response.Body)
+	return err
+}
+
+func fetchFromController(deployment DeploymentOptions, path string) error {
+	var (
+		pkgURL *url.URL
+		err    error
+	)
+	if deployment.Target.Deployment().Zone.Environment == "dev" {
+		pkgURL, err = deployment.url(fmt.Sprintf("/application/v4/tenant/%s/application/%s/instance/%s/job/%s/package",
+			deployment.Target.Deployment().Application.Tenant,
+			deployment.Target.Deployment().Application.Application,
+			deployment.Target.Deployment().Application.Instance,
+			deployment.Target.Deployment().Zone.Environment+"-"+deployment.Target.Deployment().Zone.Region,
+		))
+	} else {
+		pkgURL, err = deployment.url(fmt.Sprintf("/application/v4/tenant/%s/application/%s/package",
+			deployment.Target.Deployment().Application.Tenant,
+			deployment.Target.Deployment().Application.Application),
```

Comment:
Not including the `build` option is fine with me here.

Reply:
Agree, we can add if/when someone wants it.